django/db/backends/postgresql/schema.py

```diff
@@ -15,6 +15,10 @@ class DatabaseSchemaEditor(BaseDatabaseSchemaEditor):
     sql_create_varchar_index = "CREATE INDEX %(name)s ON %(table)s (%(columns)s varchar_pattern_ops)%(extra)s"
     sql_create_text_index = "CREATE INDEX %(name)s ON %(table)s (%(columns)s text_pattern_ops)%(extra)s"
 
+    # Setting the constraint to IMMEDIATE makes sure any deferred checks are
+    # run before attempting to drop it which is disallowed on PosrgreSQL.
```

Comment:
"makes sure any deferred checks..." -> "runs any deferred checks to allow dropping it in the same transaction."